tutorials/image_captioning/README.md

```diff
@@ -0,0 +1,138 @@
+        --output training_set.features \
+        --device-id 0 \
+        --batch-size 128 \
+        --source-image-size 3 224 224 \
+        --image-encoder-model-path /path/to/mxnet/model/filename_prefix \
+        --image-encoder-layer stage4_unit3_conv3
+
+> python -m sockeye.image_captioning.extract_features \
+        --image-root /path/to/image/dataset/folder/ \
+        --input validation_set.images \
+        --output-root /path/to/feature/cache/folder/ \
+        --output validation_set.features \
+        --device-id 0 \
+        --batch-size 128 \
+        --source-image-size 3 224 224 \
+        --image-encoder-model-path /path/to/mxnet/model/filename_prefix \
+        --image-encoder-layer stage4_unit3_conv3
+```
+In the option `--image-encoder-model-path`, `filename_prefix` should be the prefix of the MXNet mode without `-symbol.json` or `-0000.params`.
+
+The script above will generate the features stored in `/path/to/feature/cache/` and a file `training_set.features` which contains the path to the features relative to `/path/to/feature/cache/`.
+Note that finetuning of the image model is not supported yet.
+
+
+Now we can train an one-layer LSTM with attention for image captioning model as follows:
+```bash
+> python -m sockeye.image_captioning.train \
+        --source-root /path/to/feature/cache/folder/ \
+        --source training_set.features \
+        --target training_set.captions \
+        --validation-source-root /path/to/feature/cache/folder/ \
+        --validation-source validation_set.features \
+        --validation-target validation_set.captions \
+        --batch-size 64 \
+        --initial-learning-rate 0.0003 \
+        --gradient-clipping-threshold 1.0 \
+        --bucket-width 5 \
+        --max-seq-len 1:60 \
+        --fill-up replicate \
+        --output models/ \
+        --encoder image-pretrain-cnn \
+        --rnn-num-hidden 512 \
+        --rnn-decoder-state-init zero \
+        --checkpoint-frequency 200 \
+        --weight-normalization
+```
+Use the option `--load-all-features-to-memory` to load all the features to memory. This is possible depending on the size of the dataset/features and amount of available CPU memory.
+There is an initial overhead to load the feature (training does not start immediately), but with the big advantage that training is 15X-20X faster.
+
+You can add the options `--decode-and-evaluate 200 --max-output-length 60` to perform captioning of the part of the validation set (200 samples in this case) during training.
+
+### Image to Text
+
+Assuming that features were pre-extracted, you can do image captioning as follows:
+
+```bash
+> python -m sockeye.image_captioning.captioner \
+        --models models/ \
+        --input validation_set.features \
+        --source-root /path/to/feature/cache/folder/ \
+        --max-output-length 60 \
+        --batch-size 1024 \
+        --chunk-size 2048 \
+        --beam-size 3 >> validation_set.predictions
```

Comment:
'>>' might be confusing for some people.

Reply:
I'll add a brief explanation.

Reply:
Actually it is better to use `>`